attest/verifier/build.rs

```diff
@@ -290,4 +81,170 @@ fn main() {
+    write(root_anchor_path, &root_cert_pem).expect("Unable to write root anchor");
+
+    // IAS SIGNER CERT
+    let signer_key = SigningKey::random(&mut *RNG.lock().expect("mutex poisoned"));
+    let der_signer_key = signer_key
+        .to_pkcs8_der()
+        .expect("Could not export privkey to DER");
+    write(
+        signer_key_path,
+        der_signer_key
+            .to_pem("PRIVATE KEY", LineEnding::LF)
+            .expect("Could not encode signer key to PEM"),
+    )
+    .expect("Could not write signer key PEM to file");
+
+    let mut signer_subject_key = Pk::from_private_key(der_signer_key.as_bytes(), None)
+        .expect("Could not parse privkey from DER");
+
+    let signer_not_before = Time::new(
+        u16::try_from(start_time.year()).expect("Year not a u16"),
+        u8::try_from(start_time.month()).expect("Month not a u8"),
+        u8::try_from(start_time.day()).expect("Day not a u8"),
+        u8::try_from(start_time.hour()).expect("Hour not a u8"),
+        u8::try_from(start_time.minute()).expect("Minute not a u8"),
+        u8::try_from(start_time.second()).expect("Second not a u8"),
+    )
+    .expect("Could not create a not_before time");
+    let signer_not_after = Time::new(
+        u16::try_from(end_time.year()).expect("Year not a u16"),
+        u8::try_from(end_time.month()).expect("Month not a u8"),
+        u8::try_from(end_time.day()).expect("Day not a u8"),
+        u8::try_from(end_time.hour()).expect("Hour not a u8"),
+        u8::try_from(end_time.minute()).expect("Minute not a u8"),
+        u8::try_from(end_time.second()).expect("Second not a u8"),
+    )
+    .expect("Could not create a not_after time");
+
+    serial[0] += 1;
+
+    let mut builder = Builder::new();
+    let signer_cert_pem = builder
+        .subject_with_nul(SIGNER_SUBJECT)
+        .expect("Could not set subject")
+        .issuer_with_nul(ROOT_SUBJECT)
+        .expect("Could not set issuer")
+        .basic_constraints(false, None)
+        .expect("Could not set basic constraints")
+        .key_usage(KeyUsage::DIGITAL_SIGNATURE | KeyUsage::NON_REPUDIATION)
+        .expect("Could not set key usage")
+        .validity(signer_not_before, signer_not_after)
+        .expect("Could not set time validity range")
+        .serial(&serial[..])
+        .expect("Could not set serial number")
+        .subject_key(&mut signer_subject_key)
+        .issuer_key(&mut signer_issuer_key)
+        .signature_hash(HashType::Sha256)
+        .write_pem_string(&mut csprng)
+        .expect("Could not create PEM string of certificate");
+
+    write(chain_path, root_cert_pem + &signer_cert_pem).expect("Unable to write cert chain");
+}
+
+/// Returns true of the build script should generate the sim files
+fn should_generating_sim_files(data_path: impl AsRef<Path>) -> bool {
+    let data_path = data_path.as_ref();
+    let signal_file = data_path.join("delete_me_to_regenerate_sim_files.empty");
```

Comment:
Can't we regenerate if one of the generated sim files are deleted?

Reply:
Reverted back to regenerating if any of the sim files are missing, or older than 1 week